File: api/src/main/java/net/blueshell/api/dto/recovery/UserActivationRequest.java
package net.blueshell.api.dto.recovery;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;

@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
@Data
@Schema(name = "UserActivationRequest")
public class UserActivationRequest extends BaseDTO {
    @NotBlank
    private String token;
}

File: api/src/main/kotlin/net/blueshell/api/dto/recovery/UserActivationRequest.kt
package net.blueshell.api.dto.recovery

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotBlank
import net.blueshell.api.base.BaseDTO
@Schema(name = "UserActivationRequest")
class UserActivationRequest : BaseDTO() {
    @NotBlank
    val token: @NotBlank String? = null
}
